civihr_employee_portal/src/View/AbstractView.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace Drupal\civihr_employee_portal\View;
+
+abstract class AbstractView {
+
+  protected static $name = NULL;
+
+  /**
+   * @param \view $view
+   * @param \views_plugin_query_default $query
+   */
+  public function alter(&$view, &$query) {
+    // do nothing
```

Comment:
I thought about that, but it might be that you have a view that implements any one of the other [view hooks](https://api.drupal.org/api/views/views.api.php/group/views_hooks/7.x-3.x) but doesn't use `alter`, so I thought it best to leave it open. Then the method can always be called, so you could do something later like 

```
foreach ($allViewClasses as $viewClass) {
  $viewClass->alter($view, $query);
}
```

But only those that actually make changes would alter the view  or query.